### DISCUSSION TITLE
IOS Chrome can not set a string and returns nothing

### DISCUSSION THREAD

**Discussion by siavashdelkhosh81** [pmndrs/zustand#1169] [Q&A]
Hi, I am using zustand V3 with react V18 and typescript, the problem is on IOS platforms using chrome and google 
the code that set an access token is not working

here is my code

`
if (typeof data === "string") {
            setLoading(false);
            setAccess(data);
            setIsLogIn(true);
            navigate("profile");
          } else {
            alert("Wallet address and telegram Id is used before");
            window.location.reload();
          }
`
in the above code I am trying to send an request to the server to get the access token and want to save that token to the `access` state and use it everywhere 

also should mention that I am storing the token to the cookies as well

`store.ts`

`
import create from "zustand";

interface USER {
  access: string;
  setAccess: (access: string) => void;
  setIsLogIn: (data: boolean) => void;
  isLoggedIn: boolean;
  onLogin: boolean;
  setOnLogin: (data: boolean) => void;
  ref: string | null;
  setRef: (ref: string) => void;
}

export const useLogin = create<USER>((set) => ({
  access: "",
  isLoggedIn: false,
  onLogin: true,
  ref: null,

  setRef: (ref) => set((state) => ({ ...state, ref })),
  setAccess: (token) => set((state) => ({ ...state, access: token })),
  setIsLogIn: (data) => set((state) => ({ ...state, isLoggedIn: data })),

  setOnLogin: (data) => set((state) => ({ ...state, onLogin: data })),
}));
`

the code above is ok when using any other platforms such as macOS and android and even safari in IOS
but not saving the `access` on IOS search engines such as firefox chrome and google

**Comment by dai-shi** [comment]
Can anyone help?

**Reply by siavashdelkhosh81** [reply]
I have checked everything about chrome google and firefox to see why only in these apps the problem shows it self but still no answer

**Comment by donsn** [comment]
2 years later, same issue

**Reply by kcarrillor01** [reply]
same 2 years, same issue, IDK what to do

**Reply by dbritto-dev** [reply]
@donsn @kcarrillor01 would you mind sharing a minimal repro on stackblitz? I'd be happy to help
